### PR TITLE
[MSHARED-1382] support a possibility to disable snapshots update.

### PR DESCRIPTION
**Pull request by lrozenblyum**
Added a possibility to request Maven not to execute the snapshots updates.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

### CODE REVIEW

File: src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java [hunk 1]
```diff
@@ -299,6 +299,33 @@ public void testShouldSetUpdateSnapshotsFlagFromRequest() {
         assertArgumentsPresent(cli, Collections.singleton("-U"));
     }
 
+    // JUnit5: test methods don't need to be public
```
Comment (slawekjaranowski) [inline src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java:302]: you can drop `public` modifier from other methods also 😄 
now on separate PR

**Comment by slawekjaranowski** [comment]
@lrozenblyum - pleach check build result 😄 the same should be when you build locally

**Comment by lrozenblyum** [comment]
@slawekjaranowski thx I'll check it.
`mvn verify` was green at my PC (and `mvn -Prun-its verify` which is suggested by README issues a warning about a not-existing profile).
I'll check what is run in CI and simulate it locally.

**Comment by lrozenblyum** [comment]
@slawekjaranowski fixed. The cause of the difference was JAVA_HOME pointing to JDK1.8, but we need JDK17 for building as in CI to get the `spotless` errors.

File: src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java (hunk 1 above)
Comment (lrozenblyum) [reply src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java:302]: Actually the `test` prefix is also redundant for `@Test` methods, yet here I kept the project-wide convention...

**Pull request merged**